File: NancyDemo.VB/MyBootStrapper.vb
Imports Nancy.Bootstrapper
Imports Nancy.Authentication.Forms
Imports Nancy
Imports Nancy.TinyIoc
Imports NancyDemo.VB.Services
Imports NancyDemo.VB.Security

Public Class MyBootStrapper
    Inherits DefaultNancyBootstrapper

    Protected Overrides Sub ConfigureApplicationContainer(container As TinyIoCContainer)
        container.Register(Of BushService).AsSingleton()
        container.Register(Of TreeService).AsSingleton()
        container.Register(Of UserService).AsSingleton()
    End Sub

    Protected Overrides Sub ConfigureRequestContainer(container As TinyIoCContainer, context As NancyContext)
        MyBase.ConfigureRequestContainer(container, context)
        container.Register(Of IUserMapper, FakeUserMapper)()
    End Sub

    Protected Overrides Sub RequestStartup(container As TinyIoCContainer, pipelines As IPipelines, context As NancyContext)
        Dim formsAuthConfiguration = New FormsAuthenticationConfiguration With
        {
            .RedirectUrl = "~/login",
            .UserMapper = container.Resolve(Of IUserMapper)()
        }
        FormsAuthentication.Enable(pipelines, formsAuthConfiguration)
    End Sub
End Class

File: NancyDemo.Csharp/MyBootStrapper.cs
using Nancy;
using Nancy.Authentication.Forms;
using Nancy.Bootstrapper;
using Nancy.TinyIoc;
using NancyDemo.Csharp.Processors;
using NancyDemo.Csharp.Security;
using NancyDemo.Csharp.Services;

namespace NancyDemo.Csharp
{
    public class MyBootStrapper : DefaultNancyBootstrapper
    {

        protected override void ConfigureApplicationContainer(TinyIoCContainer container)
        {
            container.Register<BushService>().AsSingleton();
            container.Register<TreeService>().AsSingleton();
            container.Register<UserService>().AsSingleton();
        }

        protected override void ConfigureRequestContainer(TinyIoCContainer container, NancyContext context)
        {
            container.Register<IUserMapper, FakeUserMapper>();
        }

        protected override void RequestStartup(TinyIoCContainer container, IPipelines pipelines, NancyContext context)
        {
            var formsAuthConfiguration = new FormsAuthenticationConfiguration
                {
                    RedirectUrl = "~/login",
                    UserMapper = container.Resolve<IUserMapper>()
                };
            FormsAuthentication.Enable(pipelines, formsAuthConfiguration);
        }
    }
}
